service/controller/internal/changedetection/tcnp.go

```diff
@@ -93,6 +94,33 @@ func (t *TCNP) ShouldUpdate(ctx context.Context, md infrastructurev1alpha2.AWSMa
 		t.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("detected tenant cluster node pool should update due to operator version changes from %#q to %#q", cc.Status.TenantCluster.OperatorVersion, key.OperatorVersion(&md)))
 		return true, nil
 	}
+	if !securityGroupEqual {
+		t.logger.LogCtx(ctx, "level", "debug", "message", "detected tenant cluster node pool should update due to node pool security groups")
+		return true, nil
+	}
 
 	return false, nil
 }
+
+func securityGroupListEqual(currentSecurityGroupIDs []string, desiredSecurityGroups []string) bool {
+
+	if len(currentSecurityGroupIDs) != len(desiredSecurityGroups) {
+		return false
+	}
+
+	// lets iterate over all node pools general security groups and see if all of them are included in current security groupID list
+	for _, securityGroupID := range desiredSecurityGroups {
+		found := false
+		// try find if node pool security ID in the current security id list
+		for i := 0; i < len(currentSecurityGroupIDs); i++ {
+			if securityGroupID == currentSecurityGroupIDs[i] {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
We could heavily simplify this by sorting both lists and compare using the reflect package. 

```
func securityGroupListEqual(cur []string, des []string) bool {
    sort.Strings(cur)
    sort.Strings(des)

    if reflect.DeepEqual(cur, des) {
        return true
    } 

    return false
}
```

Reply:
I did this.